### PR TITLE
go1.12.4 fmt

### PR DESCRIPTION
**Pull request by heronhaye**
Checked out master, switched to 1.12.4, ran make fmt, make lint, and make vet.
Seems new vet rules have been added.

```
go vet ./...
# github.com/keybase/client/go/chat/signencrypt/example
../../code/src/github.com/keybase/client/go/chat/signencrypt/example/main.go:25:3: fail call has possible formatting directive %s
# github.com/keybase/client/go/libkb
../../code/src/github.com/keybase/client/go/libkb/passphrase_stream.go:124:9: Sprintf format %x has arg ps.LksClientHalf() of wrong type github.com/keybase/client/go/libkb.LKSecClientHalf
../../code/src/github.com/keybase/client/go/libkb/passphrase_stream_test.go:68:4: Errorf format %x has arg dk.LksClientHalf() of wrong type github.com/keybase/client/go/libkb.LKSecClientHalf
# github.com/keybase/client/go/service
../../code/src/github.com/keybase/client/go/service/account.go:115:2: struct field Status repeats json tag "status" also at ../libkb/api.go:45
make: *** [Makefile:8: vet] Error 2
```

### CODE REVIEW

**Comment by heronhaye** [comment]
todo: copy over git hash into pre-commit config after merging https://github.com/keybase/pre-commit-golang/pull/3

**Comment by jzila** [comment]
@heronhaye I think you need to `make` using go 1.12.4 in the protocol directory.

**Comment by heronhaye** [comment]
@jzila hm, I did that but didn't get a diff.

**Comment by jzila** [comment]
@heronhaye maybe you need to reinstall the binaries.

**Comment by jzila** [comment]
nevermind, it's failing just because CI isn't up to date.

**Comment by jzila** [comment]
I just triggered a rebuild of this with the new instances. Hopefully it passes!

File: go/kbhttp/srv.go [hunk 1]
```diff
@@ -103,6 +104,7 @@ func NewSrv(log logger.Logger, listenerSource ListenerSource) *Srv {
 	return &Srv{
 		log:            log,
 		listenerSource: listenerSource,
+		doneCh:         make(chan struct{}),
```
Comment (joshblum) [inline go/kbhttp/srv.go]: should this be initialized in the `Start` method so the server could be started/stopped multiple times?

File: go/kbhttp/srv.go [hunk 2]
```diff
@@ -152,11 +155,12 @@ func (h *Srv) Addr() (string, error) {
...
 // Stop stops listening on the server's listener source.
-func (h *Srv) Stop() {
+func (h *Srv) Stop() <-chan struct{} {
 	h.Lock()
 	defer h.Unlock()
 	if h.server != nil {
 		h.server.Close()
 		h.server = nil
 	}
+	return h.doneCh
```
Comment (joshblum) [inline go/kbhttp/srv.go]: this will hang if the server never started

Comment (jzila) [reply go/kbhttp/srv.go]: That should be fine. Right now we only block in tests.

File: go/kbhttp/srv.go (hunk 1 above)
Comment (jzila) [reply go/kbhttp/srv.go]: Is that a usage pattern we've ever done? I can make it do that but it won't address the blocking on a nil channel.

Comment (joshblum) [reply go/kbhttp/srv.go]: we usually have that flow in chat so it stuck out

File: go/kbhttp/srv.go (hunk 2 above)
Comment (joshblum) [reply go/kbhttp/srv.go]: could close before returning if server is nil

**Comment by jzila** [comment]
@keybase/kbfs-hackers PTAL at the KBFS changes.

**Pull request merged**